src/main.c

```diff
@@ -436,6 +442,23 @@ void on_keyboard_event(struct keyboard_collection* collection,
 	// TODO handle multiple symbols
 	xkb_keysym_t symbol = xkb_state_key_get_one_sym(keyboard->state, key);
 
+	if (symbol == XKB_KEY_F12) {
+		if (!is_pressed) {
+			if (inhibitor != NULL){
+				zwlr_input_inhibitor_v1_destroy(inhibitor);
+				inhibitor = NULL;
+			}
+			else {
+				inhibitor = zwlr_input_inhibit_manager_v1_get_inhibitor(input_inhibit_manager);
+			}
+		}
+		return;
+	}
+
+	if (inhibitor == NULL){
+		return;
+	}
+
```

Comment:
TBH I don't really like these conditions inside each other, but I didn't find a better logic to implement it...

Reply:
If you create an "input inhibitor module", you may find that this turns out nicer. It's fine to create an abstraction for it even though it's small. You can look at how the keyboard, pointer and seat modules are structured.